ui/main/src/app/server/angular.server..ts

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * See AUTHORS.txt
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of the OperatorFabric project.
+ */
+
+import {ServerResponse, ServerResponseStatus} from 'app/business/server/serverResponse';
+import {catchError, map, Observable, of} from 'rxjs';
+
+export class AngularServer {
+    protected processHttpResponse<dataType>(httpResponse: Observable<dataType>): Observable<ServerResponse<dataType>> {
+        return httpResponse.pipe(
+            map((data) => {
+                return new ServerResponse(data, ServerResponseStatus.OK, '');
+            }),
+            catchError((error) => {
+                let serverStatus = ServerResponseStatus.UNKNOWN_ERROR;
+                if (error.status === 404) serverStatus = ServerResponseStatus.NOT_FOUND;
+                else if (error.status === 403) serverStatus = ServerResponseStatus.FORBIDDEN;
+
```

Comment:
I wonder if it could be useful to log the error status when not  404 or 403

Reply:
the log is already present in the network panel so i do not think we have to repeat it here